superset-frontend/spec/javascripts/dashboard/util/isValidChild_spec.ts

```diff
@@ -136,20 +136,22 @@ describe('isValidChild', () => {
     invalidExamples.forEach((example, exampleIdx) => {
       let childDepth = 0;
       example.forEach((childType, i) => {
-        const shouldTestChild = Array.isArray(childType);
-
-        if (i > 0 && shouldTestChild) {
+        // should test child
+        if (i > 0 && Array.isArray(childType)) {
           const parentDepth = childDepth - 1;
           const parentType = example[i - 1];
 
+          if (typeof parentType !== 'string')
+            throw TypeError('parent must be string');
```

Comment:
Type narrowing needed due to mixed `string` and `string[]` in the array, this should never throw